src/realm/table.hpp

```diff
@@ -1161,6 +1161,20 @@ class Table {
     const BacklinkColumn& get_column_backlink(size_t ndx) const noexcept;
     BacklinkColumn& get_column_backlink(size_t ndx);
 
+    void verify_column(size_t col_ndx, const ColumnBase* col) const
+    {
+        if (col_ndx < m_cols.size()) {
+            if (m_cols[col_ndx] == col)
+                return;
+            // A column could have been inserted. Check if col can be found further on
```

Comment:
Theoretically, if there was a column delete or column swap operation, the column could have moved downwards instead of up. Can you check the whole list?